lib-impl/src/main/java/org/ergoplatform/appkit/impl/BlockchainContextImpl.java

```diff
@@ -149,5 +149,17 @@ public ErgoContract compileContract(Constants constants, String ergoScript) {
                         template.getEncodedBytes());
         return getInputBoxes(boxes);
     }
+
+    @Override
+    public Optional<InputBox> getBoxByIdFromExplorer(ErgoId boxId) {
+        Optional<TransactionOutput> res;
+        try {
+            res = Optional.ofNullable(ExplorerFacade
+                    .transactionsBoxesIdGet(_retrofitExplorer, boxId.toString()));
+        } finally {
+            res = Optional.empty();
+        }
+        return res.flatMap(out -> getInputBoxes(Collections.singletonList(out)).stream().findFirst());
```

Comment:
This doesn't seem right, all this logic should happen on the explorer side, so that unnecessary data is not leaving postgres.

Reply:
Well, I don't like exposing this inner machinery either. Maybe it should be thrown away in favor of https://github.com/ergoplatform/explorer-back/issues/232.